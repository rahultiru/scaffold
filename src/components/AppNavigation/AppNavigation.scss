.AppNavigation {
  display: flex;
  height: 100%;
  flex-direction: column;
  background-color: var(--appNavigationBackgroundColor);
}

.AppNavigation__header {
  flex: 0 auto;
}

.AppNavigation__body {
  flex: 1 100%;
  -webkit-overflow-scrolling: touch;
  overflow-y: scroll;
}

@media (--medium-screen) {
  .AppNavigation__body {
    display: block;
  }
}
