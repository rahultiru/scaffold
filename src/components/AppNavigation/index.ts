export { default } from './AppNavigation';
export * from './AppNavigation';
