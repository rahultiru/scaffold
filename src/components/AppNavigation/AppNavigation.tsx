import React from 'react';

import './AppNavigation.scss';

/**
 * Main navigation of the app.
 */
const AppNavigation: React.SFC = () => <nav className="AppNavigation" role="navigation" />

export default AppNavigation;
