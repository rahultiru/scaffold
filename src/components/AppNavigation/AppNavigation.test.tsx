jest.mock('components/NavigationGroups', () => 'LoadableNavigationGroups');
jest.mock('components/GlobalSearchBox', () => 'LoadableGlobalSearchBox');

import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import AppNavigation from '.';

describe('AppNavigation', () => {
  let component: ShallowWrapper<{}>;

  beforeEach(() => {
    component = shallow(<AppNavigation />);
  });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });

  describe('body content', () => {
    it('renders', () => {});
  });
});
