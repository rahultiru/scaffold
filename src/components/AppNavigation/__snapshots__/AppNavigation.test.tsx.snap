// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation body content renders 1`] = `
<div
  className="AppNavigation__body"
>
  <Block
    textSize="mediumSub"
  >
    <LoadableNavigationGroups />
  </Block>
</div>
`;

exports[`AppNavigation renders 1`] = `
<nav
  className="AppNavigation"
  role="navigation"
>
  <div
    className="AppNavigation__header"
  >
    <Block
      padding="large"
      textSize="mediumSub"
      topSpacing="xSmall"
    >
      <div
        className="AppNavigation__header"
      >
        <AppNavigationButtons />
        <Block
          topSpacing="large"
        >
          <LoadableGlobalSearchBox />
        </Block>
      </div>
    </Block>
  </div>
  <ResponsiveRender
    breakpoint={1024}
    renderAbove={[Function]}
  />
</nav>
`;
